<launch>
  <arg name="role_name" default="hero" />
  <arg name="control_loop_rate" default="0.05" />
  <arg name="view_camera" default="False" />
  <arg name="camera_resolution" default="1280" />

  <node pkg="perception" type="vision_node.py" name="VisionNode" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="view_camera" value="$(arg view_camera)" />
    <param name="camera_resolution" value="$(arg camera_resolution)"/>
    <rosparam file="$(find perception)/config/trafficlight_config.yaml" command="load" />

    <!--
        Image-Segmentation:
      - yolov8x-seg
      - yolo11n-seg
      - yolo11s-seg
      - yolo11m-seg
      - yolo11l-seg
      -->

    <param name="model" value="yolo11m-seg" />
  </node>

  <node pkg="perception" type="traffic_light_node.py" name="TrafficLightNode" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="side" value="Center" />
    <param name="model" value="/workspace/code/perception/src/traffic_light_detection/models/model_acc_92.48_val_91.88.pt" />
    <param name="tfs_debug" value="False"/>
  </node>

  <node pkg="perception" type="lidar_distance.py" name="lidar_distance" output="screen">
    <!-- Cluster configuration -->
    <param name="clustering_lidar_z_min" value="-1.4" />
    <param name="dbscan_eps" value="0.4" />
    <param name="dbscan_min_samples" value="10" />
  </node>

  <node pkg="perception" type="Lanedetection_node.py" name="Lanedetection" output="screen">
    <param name="role_name" value="$(arg role_name)" />
  </node>

  <node pkg="perception" type="radar_node.py" name="radar_node" output="screen">
    <param name="dbscan_eps" value="0.3"/>
    <param name="dbscan_samples" value="1"/>
    <param name="data_buffered" value="False"/>
    <param name="data_buffer_time" value="0.1"/>
    <param name="imu_debug" value="False"/>
    <param name="accelerometer_arrow_size" value="2"/>
    <param name="accelerometer_factor" value="0.05"/>
    <param name="enable_clustering" value="True"/>
    <param name="enable_debug_info" value="False"/>
  </node>


  <node pkg="perception" type="lane_position.py" name="lane_position" output="screen">
    <param name="epsilon" value="1.8"/>
    <param name="min_samples" value="4"/>
    <param name="confidence_threshold" value="0.6"/>
    <param name="angle_weight" value="0.3"/>
    <param name="size_weight" value="0.3"/>
    <param name="std_dev_weight" value="0.4"/>
    <param name="angle_normalization" value="25.0"/>
    <param name="size_normalization" value="15.0"/>
    <param name="std_dev_normalization" value="0.1"/>
    <param name="angle_prediction_threshold" value="5.0"/>
    <param name="y_tolerance" value="1.0"/>
    <param name="line_length" value="15.0"/>
    <param name="line_width" value="0.5"/>
  </node>

</launch>
